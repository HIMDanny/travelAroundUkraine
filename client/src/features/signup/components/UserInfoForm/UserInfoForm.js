import React from 'react';
import { Field } from 'formik';
import { TextField, RadioGroup } from 'formik-mui';
import { DatePicker } from 'formik-mui-x-date-pickers';
import { Grid, Typography, FormLabel, Radio, FormControlLabel } from '@mui/material';
import { formModel } from '../../data';

const UserInfoForm = () => {
  const { firstName, lastName, gender, birthdate, telephone } = formModel;

  return (
    <>
      <Typography variant="h3" mb={3}>
        User Info
      </Typography>
      <Grid container spacing={3} sx={{ mb: 3 }}>
        <Grid item xs={12} laptop={6}>
          <Field component={TextField} name={firstName.name} label={firstName.label} fullWidth required />
        </Grid>

        <Grid item xs={12} laptop={6}>
          <Field component={TextField} name={lastName.name} label={lastName.label} fullWidth required />
        </Grid>

        <Grid item xs={12} laptop={4}>
          <FormLabel id={gender.name}>{gender.label}</FormLabel>
          <Field component={RadioGroup} aria-labelledby={gender.name} row name={gender.name}>
            <FormControlLabel
              componentsProps={{ typography: { gutterBottom: false } }}
              value="male"
              control={<Radio />}
              label="Male"
            />
            <FormControlLabel
              componentsProps={{ typography: { gutterBottom: false } }}
              value="female"
              control={<Radio />}
              label="Female"
            />
            <FormControlLabel
              componentsProps={{ typography: { gutterBottom: false } }}
              value="other"
              control={<Radio />}
              label="Other"
            />
          </Field>
        </Grid>

        <Grid item xs={12} laptop={4}>
          <Field component={DatePicker} disableFuture name={birthdate.name} label={birthdate.label} />
        </Grid>

        <Grid item xs={12} laptop={4}>
          <Field component={TextField} name={telephone.name} label={telephone.label} fullWidth />
        </Grid>
      </Grid>
    </>
  );
};

export default UserInfoForm;
