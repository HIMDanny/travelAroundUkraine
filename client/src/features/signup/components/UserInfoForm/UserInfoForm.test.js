/* eslint-disable react/prop-types */
import { render } from '@testing-library/react';
import { Formik } from 'formik';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import UserInfoForm from './UserInfoForm';

jest.mock('@mui/material/Grid', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Radio', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
jest.mock('@mui/material/FormControlLabel', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/FormLabel', () => ({ children }) => <div>{children}</div>);
jest.mock('formik-mui-x-date-pickers', () => () => <input />);

const LocalizationWrapper = ({ children }) => (
  <LocalizationProvider dateAdapter={AdapterDateFns}>{children}</LocalizationProvider>
);

const FormWrapper = ({ children }) => (
  <Formik>
    <LocalizationWrapper>{children}</LocalizationWrapper>
  </Formik>
);

describe('UserInfoForm snapshot test', () => {
  test('should UserInfoForm match snapshot', () => {
    const { asFragment } = render(
      <FormWrapper>
        <UserInfoForm />
      </FormWrapper>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
