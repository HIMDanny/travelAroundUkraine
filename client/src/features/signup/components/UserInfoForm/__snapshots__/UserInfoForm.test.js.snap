// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoForm snapshot test should UserInfoForm match snapshot 1`] = `
<DocumentFragment>
  <p>
    User Info
  </p>
  <div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div>
          First Name
        </div>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            name="firstName"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                First Name *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div>
          Last Name
        </div>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            name="lastName"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Last Name *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div>
      <div>
        Gender
      </div>
      <div
        aria-labelledby="gender"
        class="MuiFormGroup-root MuiFormGroup-row css-qfz70r-MuiFormGroup-root"
        role="radiogroup"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
    <div>
      <input
        label="Birthdate"
        name="birthdate"
        value=""
      />
    </div>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div>
          Phone
        </div>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r3:"
            name="telephone"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Phone
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
